dfastmi/batch/AnalyserDflowfm.py

```diff
@@ -0,0 +1,376 @@
+        """
+        self._logger = AnalyserDflowfmLogger(display, report)
+        self._needs_tide = needs_tide
+        self._old_zmin_zmax = old_zmin_zmax
+        self._outputdir = outputdir
+    
+    def analyse(self, 
+                q_threshold : float,
+                tstag : float,
+                discharges : Vector,
+                fraction_of_year : Vector,
+                rsigma : Vector,
+                slength : float,
+                nwidth : float,
+                ucrit : float,
+                filenames : Dict[Any, Tuple[str,str]],
+                xykm : LineString,
+                n_fields : int,
+                tide_bc : Tuple[str, ...],
+                plotops : Dict
+                ) -> Tuple[bool, OutputDataDflowfm]:
+        """
+        Perform analysis based on D-Flow FM data.
+        Read data from D-Flow FM output files and perform analysis.
+
+        Arguments
+        ---------
+        q_threshold : float
+            Threshold discharge above which the measure is active.
+        tstag : float
+            Fraction of year that the river is stagnant.
+        discharges : Vector
+            Array of discharges; one for each forcing condition. (Q list of discharges)
+        fraction_of_year : Vector
+            Fraction of year represented by each forcing condition. (T list of fraction of year)
+        rsigma : Vector
+            Array of relaxation factors; one per forcing condition.
+        slength : float
+            The expected yearly impacted sedimentation length.
+        nwidth : float
+            normal width of the reach.
+        ucrit : float
+            Critical flow velocity [m/s].
+        filenames : Dict[Any, Tuple[str,str]]
+            Dictionary of the names of the data file containing the simulation
+            results to be processed. The conditions (discharge, wave conditions,
+            ...) are the key in the dictionary. Per condition a tuple of two file
+            names is given: a reference file and a file with measure.
+        xykm : shapely.geometry.linestring.LineString
+            Original river chainage line.
+        n_fields : int
+            Number of fields to process (e.g. to cover a tidal period).
+        tide_bc : Tuple[str, ...]
+            Array of tidal boundary condition; one per forcing condition.
+        plotops : Dict
+            Dictionary of plot settings
+
+        Returns
+        -------
+        success : bool
+            Flag indicating whether analysis could be carried out.
+        Output data : OutputDataDflowfm
+            DTO with the data which is needed to create a report.
+        """
+        one_fm_filename, missing_data = self._get_first_fm_data_filename(q_threshold, discharges, rsigma, filenames, tide_bc)
+
+        if missing_data:
+            return missing_data, None
+
+        self._logger.log_load_mesh()
+        xn, yn, face_node_connectivity = self._get_xynode_connect(one_fm_filename)
```

Comment:
can this be a void method called initialize_xynode_connect setting a object in this class with xn, yn and face_node_connectivity

Reply:
that is possible, will look into it

Reply:
implemented, but resolved it in anohter way.